File: Starters/C#/4.3 - Test Test Patterns/TestingPatterns/models/ShoppingBasket.cs
using System;
using System.Collections.Generic;
using System.Text;

public class ShoppingBasket
{
    private List<Product> items = new List<Product>();

    public void add(Product product)
    {
        items.Add(product);
    }

    public double totalWithTax(double taxRate)
    {
        double total = 0;
        foreach (Product p in items)
        {
            total += p.getPrice();
        }
        return total * (1 + taxRate);
    }

    public int itemCount()
    {
        return items.Count;
    }
}

File: Starters/Java/4.3 - Testing patterns/src/models/ShoppingBasket.java
package models;

import java.util.ArrayList;
import java.util.List;

public class ShoppingBasket {
	private List<Product> items = new ArrayList<Product>();

	public void add(Product product) {
		items.add(product);
	}

	public double totalWithTax(double taxRate) {
		double total = 0;
		for (Product p : items) {
			total += p.getPrice();
		}
		return total * (1 + taxRate);
	}

	public int itemCount() {
		return items.size();
	}
}
